conformance/tests/httproute-redirect-port.go

```diff
@@ -38,70 +39,172 @@ var HTTPRouteRedirectPort = suite.ConformanceTest{
+						Path:             "/port-and-host",
+						UnfollowRedirect: true,
+					},
+					Response: http.Response{
+						StatusCode: 302,
+					},
+					RedirectRequest: &roundtripper.RedirectRequest{
+						Host: "example.org",
+						Port: "8083",
+					},
+					Namespace: ns,
+				},
 			},
-			Namespace: ns,
-		}, {
-			Request: http.Request{
-				Path:             "/port-and-status",
-				UnfollowRedirect: true,
+			{
+				expectedResponse: http.ExpectedResponse{
+					Request: http.Request{
+						Path:             "/port-and-status",
+						UnfollowRedirect: true,
+					},
+					Response: http.Response{
+						StatusCode: 301,
+					},
+					RedirectRequest: &roundtripper.RedirectRequest{
+						Port: "8083",
+					},
+					Namespace: ns,
+				},
 			},
-			Response: http.Response{
-				StatusCode: 301,
+			{
+				expectedResponse: http.ExpectedResponse{
+					Request: http.Request{
+						Path:             "/port-and-host-and-status",
+						UnfollowRedirect: true,
+					},
+					Response: http.Response{
+						StatusCode: 302,
+					},
+					RedirectRequest: &roundtripper.RedirectRequest{
+						Port: "8083",
+						Host: "example.org",
+					},
+					Namespace: ns,
+				},
 			},
-			RedirectRequest: &roundtripper.RedirectRequest{
-				Port: "8083",
+			{
+				// When requestRedirect.Port is set to 80 and the interpreted scheme is
+				// http, port 80 should not be ADDED to the response Location header.
+				// (The Location header MAY still result in having a port if the request
+				// Host header contains one.)
+				expectedResponse: http.ExpectedResponse{
+					Request: http.Request{
+						Path:             "/http-listener-on-80-and-redirect-to-80",
+						UnfollowRedirect: true,
+					},
+					Response: http.Response{
+						StatusCode: 302,
+					},
+					RedirectRequest: &roundtripper.RedirectRequest{},
+					Namespace:       ns,
+				},
 			},
-			Namespace: ns,
-		}, {
-			Request: http.Request{
-				Path:             "/port-and-host-and-status",
-				UnfollowRedirect: true,
+			{
+				// When requestRedirect.Port is not set and the interpreted scheme is
+				// http, port 80 should not be ADDED to the response Location header.
+				// (The Location header MAY still result in having a port if the request
+				// Host header contains one.)
+				expectedResponse: http.ExpectedResponse{
+					Request: http.Request{
+						Path:             "/http-listener-on-80-and-redirect-port-not-set",
+						UnfollowRedirect: true,
+					},
+					Response: http.Response{
+						StatusCode: 301,
+					},
+					RedirectRequest: &roundtripper.RedirectRequest{},
+					Namespace:       ns,
+				},
 			},
-			Response: http.Response{
-				StatusCode: 302,
+			{
+				// When requestRedirect.Port is set to 443 and the interpreted scheme is
+				// https, port 443 should not be ADDED to the response Location header.
+				// (The Location header MAY still result in having a port if the request
+				// Host header contains one.)
```

Comment:
This comment makes sense but I'm either misunderstanding the test case or it's just not testing/verifying the same level of detail as the comment suggests. The same comment applies to several test cases, though I could just be missing something obvious.

Reply:
Obsolete